lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1182,6 +1182,17 @@ def modify_connection(self):
             cmd = self.modify_connection_bridge_slave()
         elif self.type == 'vlan':
             cmd = self.modify_connection_vlan()
+            # If no dns given, then we assume that IPv4/IPv6 is not used on the interface
+            # so we are going to disable the protocol, otherwise interface won't come up.
+
+            if self.ip4 is 'disabled':
+                self.execute_command(cmd)
+                cmd = self.modify_conn_disable_ipv4()
```

Comment:
You are not executing command returned by `modify_conn_disable_ipv4` so I am sure that it has no effect.

Reply:
My bad. Didn't see if in later stages.